### PR TITLE
add endpoint, domain, controller for getting a cohort by ID

### PR DESCRIPTION
**Pull request by Alex90Jennings**

### CODE REVIEW

File: src/controllers/cohort.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
-import { createCohort, getCohorts } from '../domain/cohort.js'
+import { createCohort, getCohorts, getCohort } from '../domain/cohort.js'
 import { sendDataResponse, sendMessageResponse } from '../utils/responses.js'
 
 export const create = async (req, res) => {
-  const { cohort_name: cohortName } = req.body
```
Comment (dearshrewdwit) [inline src/controllers/cohort.js:5]: These deletions in the create func look to me incorrect - check in with your team about them

File: src/domain/cohort.js [hunk 2]
```diff
@@ -29,6 +29,15 @@ export async function getCohorts() {
   return Cohorts
 }
 
+export async function getCohort(id) {
+  id = Number(id)
+  const Cohort = await dbClient.cohort.findUnique({
```
Comment (dearshrewdwit) [inline src/domain/cohort.js:34]: Keep the variable here lower case - the capital Cohort should represent the cohort class.

Comment (dearshrewdwit) [reply src/domain/cohort.js:34]: The `findUnique` query returns `null` if the record is not found.
> https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique

This won't trigger the try/catch block in the controller, you'd simply send back a response with `null` in the data field. 

Instead, either throw an error directly from prisma using the option `RejectOnNotFound` 
> https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#options-2

or the abstracted query method `findUniqueOrThrow`
> https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#finduniqueorthrow

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: src/controllers/cohort.js (hunk 1 above)
Comment (Alex90Jennings) [reply src/controllers/cohort.js:5]: readded the code, not sure how it was deleted

File: src/domain/cohort.js (hunk 2 above)
Comment (Alex90Jennings) [reply src/domain/cohort.js]:
```js
export async function getCohort(id) {
  id = Number(id)
  const cohort = await dbClient.cohort.findUnique({
    where: { id }
  })

  if (cohort) {
    return cohort
  }

  throw new Error()
}
```
We've included a more robust error to allow for if the cohort if not found

Comment (dearshrewdwit) [reply src/domain/cohort.js]: Nice. But please use one of the two ways described in the previous comment

> The `findUnique` query returns `null` if the record is not found.
> 
> > https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique
> 
> This won't trigger the try/catch block in the controller, you'd simply send back a response with `null` in the data field.
> 
> Instead, either throw an error directly from prisma using the option `RejectOnNotFound`
> 
> > https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#options-2
> 
> or the abstracted query method `findUniqueOrThrow`
> 
> > https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#finduniqueorthrow

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
Nice - See comment

File: src/domain/cohort.js (hunk 2 above)
Comment (Alex90Jennings) [reply src/domain/cohort.js]: added rejectOnNotFound: true to avoid returning null

**Review by dearshrewdwit — APPROVED** [review]

**Comment by dearshrewdwit** [comment]
LGTM

**Review by PBara7a — APPROVED** [review]

**Pull request merged**